doc/release/upcoming_changes/28576.new_feature.rst

```diff
@@ -0,0 +1,15 @@
+Allow ``out=...`` in ufuncs to ensure array result
+--------------------------------------------------
+NumPy has the sometimes difficult behavior that it currently usually
+returns scalars rather than 0-D arrays (even if the inputs were 0-D arrays).
+This is especially problematic for non-numerical dtypes (e.g. ``object``).
+
+For ufuncs (i.e. most simple math functions) it is now possible
+to use ``out=...`` (literally `...`, e.g. ``out=Ellipsis``) which is identical in behavior to ``out`` not
+being passed, but will ensure a non-scalar return.
+This spelling is borrowed from ``arr1d[0, ...]`` where the ``...``
+also ensures a non-scalar return.
```

Comment:
Maybe we could also mention that `arr0d[...]` is a valid lvalue as in `arr0d[...] = 3.14`

Reply:
TBH, I don't really understand why `arr[...]` assignment is relevant in this context?  (Besides any excuse for telling folks about `...` is a good excuse ;).)

Reply:
You are right, my comment is not relevant.

From the observation that `arr0d[()]` is scalar and `arr0d[...]` is a 0D array I *wrongly* assumed that the first is not a valid lvalue... 

But I just realized that `arr0d.__setitem__((), 3.14)` and `arr0d.__setitem__(Ellipsis, 3.14)` are prefectly equivalent, so please forget my comment.